inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape_matmul.cpp

```diff
@@ -36,19 +37,22 @@ void dynamicToStaticShapeMatMul(std::shared_ptr<ngraph::Node> target) {
     VPU_THROW_UNLESS(matmul, "dynamicToStaticShapeMatMul transformation is not applicable for {}, it should be {} instead",
             target, ngraph::opset3::MatMul::type_info);
 
-    auto shapeToConstant = [&target](const ngraph::Output<ngraph::Node> & output) -> std::shared_ptr<ngraph::opset3::Constant> {
+    auto shapeToConstant = [&target](const ngraph::Output<ngraph::Node>& output,
+                                     const ngraph::element::Type& elementType) -> std::shared_ptr<ngraph::opset3::Constant> {
         VPU_THROW_UNLESS(output.get_partial_shape().is_static(),
                          "DynamicToStaticShape transformation for {} of type {} expects static shape on inputs without DSR",
                          target->get_friendly_name(), target->get_type_info());
-        return ngraph::opset3::Constant::create(ngraph::element::i64, {output.get_shape().size()}, output.get_shape());
+        return ngraph::opset3::Constant::create(elementType, {output.get_shape().size()}, output.get_shape());
     };
 
     const auto a_input_DSR = ngraph::as_type_ptr<ngraph::vpu::op::DynamicShapeResolver>(target->input_value(0).get_node_shared_ptr());
     const auto b_input_DSR = ngraph::as_type_ptr<ngraph::vpu::op::DynamicShapeResolver>(target->input_value(1).get_node_shared_ptr());
     VPU_THROW_UNLESS(a_input_DSR || b_input_DSR, "DynamicToStaticShape transformation for {} of type {} expects at least one DSR as input",
                      target->get_friendly_name(), target->get_type_info());
-    ngraph::Output<ngraph::Node> a_input_shape = a_input_DSR ? a_input_DSR->input_value(1) : shapeToConstant(target->input_value(0));
-    ngraph::Output<ngraph::Node> b_input_shape = b_input_DSR ? b_input_DSR->input_value(1) : shapeToConstant(target->input_value(1));
+    const auto shapeElementType = a_input_DSR ? a_input_DSR->get_input_element_type(1) : b_input_DSR->get_input_element_type(1);
```

Comment:
I'd prefer to assert `a_input_DSR->get_input_element_type(1) == b_input_DSR->get_input_element_type(1)` if both are DSR

Reply:
Done.